src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,297 @@
+    eventMapper.put("Checkout Started", Event.BEGIN_CHECKOUT);
+    eventMapper.put("Order Completed", Event.ECOMMERCE_PURCHASE);
+    eventMapper.put("Order Refunded", Event.PURCHASE_REFUND);
+    eventMapper.put("Product Viewed", Event.VIEW_ITEM);
+    eventMapper.put("Product List Viewed", Event.VIEW_ITEM_LIST);
+    eventMapper.put("Payment Info Entered", Event.ADD_PAYMENT_INFO);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+    eventMapper.put("Product Added to Wishlist", Event.ADD_TO_WISHLIST);
+    eventMapper.put("Product Shared", Event.SHARE);
+    eventMapper.put("Product Clicked", Event.SELECT_CONTENT);
+    eventMapper.put("Product Searched", Event.SEARCH);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+
+    if (eventMapper.containsKey(eventName)) {
+      eventName = eventMapper.get(eventName);
+    }
+
+    if (eventName.contains(" ")) {
+      eventName = eventName.trim().toLowerCase().replaceAll(" ", "_");
+    }
+
+    if (eventName.length() > 40) {
+      eventName = trimKey(eventName);
+    }
+
+    return eventName;
+  }
+
+  private Bundle formatProperties(Properties properties) {
+
+    if ((properties.revenue() != 0 || properties.total() != 0) && isNullOrEmpty(properties.currency())) {
+      logger.verbose(
+          "You must set `currency` in your event's property object to accurately "
+              + "pass 'value' to Firebase.");
+    }
+
+    Bundle bundle = new Bundle();
+
+    for (Map.Entry<String, Object> entry : properties.entrySet()) {
+      String property = entry.getKey();
+      property  = mapProperty(property);
+
+      if (entry.getValue() instanceof Integer) {
+        int value = (int) entry.getValue();
+        bundle.putInt(property, value);
+        logger.verbose("bundle.putInt(%s, %s);", property, value);
+      }
+
+      if (entry.getValue() instanceof Double) {
+        double value = (double) entry.getValue();
+        bundle.putDouble(property, value);
+        logger.verbose("bundle.putDouble(%s, %s);", property, value);
+      }
+
+      if (entry.getValue() instanceof Long) {
+        long value = (long) entry.getValue();
+        bundle.putLong(property, value);
+        logger.verbose("bundle.putLong(%s, %s);", property, value);
+      }
+
+      if (entry.getValue() instanceof String) {
+        String value = String.valueOf(entry.getValue());
+
+        if (ISO_DATE.matcher(value).matches()) {
+          value = value.substring(0, 10);
+        }
+
+        bundle.putString(property, value);
+        logger.verbose("bundle.putString(%s, %s);", property, value);
+      }
+
+      if (entry.getValue() instanceof Date) {
+
+        Date value = (Date) entry.getValue();
+        String formattedDate = formatDate(value);
+
+        bundle.putString(property, formattedDate);
+        logger.verbose("bundle.putString(%s, %s);", property, formattedDate);
+      }
+    }
+
+    return bundle;
+  }
+
+  private String mapProperty(String property) {
+
```

Comment:
Unnecessary blank line as per https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace. Please take a second pass through the file to see if you missed any others. I'm still seeing a few others.